### PR TITLE
api: rename Cluster kind to ScyllaCluster in missing places

### LINKED ISSUE #144: Consider using less common kind
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

I would suggest to replace wide meaning kind name  `kind: Cluster` to something less common.

Most of the operators using own kinds, eg:
- Elasticsearch operator uses `kind: Elasticsearch`
- Redis operator uses `kind: RedisCluster`
- PostgreSQL operator uses `kind: postgresql`
- Cassandra operator uses `kind: CassandraDatacenter`

Scylla-operator should use something simmilar.

**What is use case behind this feature:**

Using so wide meaning kinds is not common in Kubernetes.

**Additional Information:**

### PR DESCRIPTION
**Pull request by zimnx**
Problems solved:
 - webhooks were watching old "Cluster" resource
- controller wasn't able to update cluster status
- controller watch on owned StatefulSet events was registered on old "Cluster" resource
    
Fixes #144

### CODE REVIEW

**Comment by espindola** [comment]
Please open one issue for adding regression tests for each of the problems you found.

Some of the files in here are generated, no? Please split the patch in

* Change the non generated files
* Change to the generated files

In the second patch, say which command was used to regenerate them.

**Comment by zimnx** [comment]
> Please open one issue for adding regression tests for each of the problems you found.
> 
Added issue to implement basic scenario first. I wasn't able to spawn any node using master due to above problems. https://github.com/scylladb/scylla-operator/issues/220

> Some of the files in here are generated, no? Please split the patch in
> 
>     * Change the non generated files
> 
>     * Change to the generated files
> 
> 
> In the second patch, say which command was used to regenerate them.

done

File: examples/eks/cluster.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ rules:
   - apiGroups:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
Comment (espindola) [inline examples/eks/cluster.yaml:43]: This file is a copy and paste from the one in config, no? It seems to have some arbitrary dependencies:

````
$ diff -u  ./config/manager/cluster/cluster.yaml examples/eks/cluster.yaml
... 
apiVersion: scylla.scylladb.com/v1alpha1
 kind: Cluster
 metadata:
-  name: cluster
-  namespace: scylla-manager
+  name: scylla-cluster
+  namespace: scylla
```

File: examples/generic/cluster.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ rules:
   - apiGroups:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
Comment (espindola) [inline examples/generic/cluster.yaml:43]: and this one.

File: examples/gke/cluster.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ rules:
   - apiGroups:
       - scylla.scylladb.com
     resources:
-      - clusters
+      - scyllaclusters
```
Comment (espindola) [inline examples/gke/cluster.yaml:43]: and this one.

File: pkg/api/v1alpha1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (espindola) [inline pkg/api/v1alpha1/zz_generated.deepcopy.go:24]: According to the file name this file is generated.

File: examples/eks/cluster.yaml (hunk 1 above)
Comment (zimnx) [reply examples/eks/cluster.yaml:43]: `cluster.yaml` files are not generated. These are examples what user should provide and these were manually crafted or copied from generic if they suits.

**Comment by zimnx** [comment]
@espindola I found that autogeneration stopped working some time ago or wasn't there in some files. I fixed the issues, extracted to separate commits. PTAL

File: pkg/api/v1alpha1/cluster_webhook.go [hunk 5]
```diff
@@ -33,14 +33,11 @@ func (r *ScyllaCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (espindola) [inline pkg/api/v1alpha1/cluster_webhook.go:36]: Is this change part of running "make manifests"?

File: config/manager/rbac/role.yaml [hunk 6]
```diff
@@ -6,19 +6,25 @@ metadata:
   creationTimestamp: null
   name: manager-role
 rules:
-- resources:
```
Comment (espindola) [inline config/manager/rbac/role.yaml]: This file is modified in 3 of the 4 patches. It is auto generated, correct? If so, it should be modified in only one patch, no?

**Review by espindola — COMMENTED** [review]
Thanks a lot for fixing the Makefile.

I think this needs to be rebased on top of current master.

Auto generated files are still modified in multiple patches. Why?

File: config/manager/rbac/role.yaml (hunk 6 above)
Comment (zimnx) [reply config/manager/rbac/role.yaml]: done, somehow missed it during rebasing and squashing

File: pkg/api/v1alpha1/cluster_webhook.go (hunk 5 above)
Comment (zimnx) [reply pkg/api/v1alpha1/cluster_webhook.go:36]: no, i manually removed this comment. It was generated by the `kubebuilder` during project initializaiton.

**Review by espindola — APPROVED** [review]
LGTM

**Pull request merged**